doc/changes/DM-34696.feature.rst

```diff
@@ -0,0 +1 @@
+* Added workarounds for mypy errors in `lsst.pipe.base.Struct` and `lsst.pipe.base.PipelineTask`.
```

Comment:
I'm not sure I'd call this a feature that we want to headline. It seems more like "misc" to me.